control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -1048,3 +1059,52 @@ impl CreateVolumeExeVal for CreateVolumeSource<'_> {
         }
     }
 }
+
+// spread_label_is_same checks if the node labels is same as that of all existing replicas.
+fn spread_label_is_same(
+    replica: &Replica,
+    replica_candidate: &CreateReplica,
+    context: &Context,
+) -> Result<bool, SvcError> {
+    let Some(ref topology) = context.volume.as_ref().topology else {
+        return Ok(false);
+    };
+
+    let Some(node_topology) = topology.node.as_ref() else {
+        return Ok(false);
+    };
+    let node_exclusion_label_match = match node_topology {
+        NodeTopology::Explicit(_) => true,
+        NodeTopology::Labelled(labelled_topology) => {
+            // If the NodeSpreadTopologyKey is present in the Volume Node Topology, then
+            // get the node details from replica candidate and
+            // check if the node labels is same as that of all existing replicas.
+            // If its same then don't consider the pool for replica creation
+            let mut exclusion_label_match = false;
+            if !labelled_topology.exclusion.is_empty() {
+                let candidate_node = context
```

Comment:
Can't we use the node from the candidate itself?

Reply:
No, the node in candidate replica is just the `NodeId` . 
```
pub struct CreateReplica {
    /// Id of the io-engine instance.
    pub node: NodeId,

Reply:
Why do we need a new type. IMHO the present approach is good enough, Its just that we need the node id to get the labels. 

If you are not convinced , can work on that on another PR, would request you to let this feature in first.